deps/rabbit/src/rabbit_mirror_queue_slave.erl

```diff
@@ -563,21 +563,24 @@ send_mandatory(#delivery{mandatory  = true,
 
 send_or_record_confirm(_, #delivery{ confirm = false }, MS, _State) ->
     MS;
-send_or_record_confirm(published, #delivery { sender     = ChPid,
+send_or_record_confirm(Status, #delivery { sender     = ChPid,
                                               confirm    = true,
                                               msg_seq_no = MsgSeqNo,
-                                              message    = #basic_message {
-                                                id            = MsgId,
-                                                is_persistent = true } },
-                       MS, #state{q = Q}) when ?amqqueue_is_durable(Q) ->
-    maps:put(MsgId, {published, ChPid, MsgSeqNo} , MS);
-send_or_record_confirm(_Status, #delivery { sender     = ChPid,
-                                            confirm    = true,
-                                            msg_seq_no = MsgSeqNo },
-                       MS, #state{q = Q} = _State) ->
-    ok = rabbit_classic_queue:confirm_to_sender(ChPid,
-                                                amqqueue:get_name(Q), [MsgSeqNo]),
-    MS.
+                                              message    = Msg
+                                            },
+                       MS, #state{q = Q}) ->
+    MsgId = mc:get_annotation(id, Msg),
+    IsPersistent = mc:is_persistent(Msg),
+    case IsPersistent of
+        true when ?amqqueue_is_durable(Q) andalso
+                  Status == published ->
+            maps:put(MsgId, {published, ChPid, MsgSeqNo}, MS);
+        _ ->
+            ok = rabbit_classic_queue:confirm_to_sender(ChPid,
+                                                        amqqueue:get_name(Q),
+                                                        [MsgSeqNo]),
+            MS
+    end.
```

Comment:
I think the change here is correct but considering mirrored queues are going away soon(tm) I would prefer a simpler change that leaves the two clauses there was before, so as to minimize the risk.

Reply:
I agree but the previous code relies on pattern matching in the function head which with `mc` we can no longer do so we'd need the ispersistent + is durable check in both clauses in that case.

Reply:
OK.